app/forms/waste_carriers_engine/registration_number_form.rb

```diff
@@ -4,11 +4,13 @@ module WasteCarriersEngine
   class RegistrationNumberForm < BaseForm
     attr_accessor :company_no, :business_type
 
+    validates :company_no, "defra_ruby/validators/companies_house_number": true
+
     def initialize(transient_registration)
       super
-      self.company_no = @transient_registration.company_no
+      self.company_no = transient_registration.company_no
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.company_no = transient_registration.company_no
```